optuna/samplers/_cmaes.py

```diff
@@ -387,7 +442,7 @@ def sample_relative(
         study._storage.set_trial_system_attr(
             trial._trial_id, generation_attr_key, optimizer.generation
         )
-        study._storage.set_trial_system_attr(trial._trial_id, "cma:n_restarts", n_restarts)
```

Comment:
In the implementation of the master branch, when `use_separable_cma` is True, the keys for reading and writing n_restarts do not seem to be the same. Is this expected behavior?

Reply:
I'm not confident about expected behavior, but it seems to be weird that the `get` and `set` key is different.

In this case, this `set_trial_system_attr` will set the storage's `trial.system_attr`, and `get` (`_cmaes.py:412`) is getting from the same `trial.system_attr`, and restore optimizer.

If using `SEPCMA` (by setting `use_separable_cma=False`), trials were saved to `system_attr` with a key starts with `cma:` and restored with `sepcma:` so it seems this code (`_cmaes.py L417-424`)
```
            if not self._use_separable_cma and "cma:optimizer" in optimizer_attrs:
                # Check "cma:optimizer" key for backward compatibility.
                optimizer_str = optimizer_attrs["cma:optimizer"]
            else:
                optimizer_str = _concat_optimizer_attrs(optimizer_attrs)

            n_restarts: int = trial.system_attrs.get(attr_key_n_restarts, 0)
            return pickle.loads(bytes.fromhex(optimizer_str)), n_restarts
```
never be executed with `use_separable_cma=False`

Reply:
And also it seems to be natural to branch [this attribute](https://github.com/zaburo-ch/optuna/blob/cmaes-group-decomposition-search-space/optuna/samplers/_cmaes.py#L344-L347) corresponding to `SepCMA` and `CMA`.